### PR TITLE
Feature/Planet

### PR DESCRIPTION
**Pull request by xhad**
Added Planet Page
- proposed context state management style ( please give comments and suggest a better way/improvements)
- Need to do something to manage global state in a store
- need to figure out syles

### CODE REVIEW

File: packages/client/src/index.js [hunk 1]
```diff
@@ -3,10 +3,14 @@ import ReactDOM from 'react-dom'
 import './shared/styles/index.css'
 import Routes from './root-routes'
 import * as serviceWorker from './serviceWorker'
+import { PlanetProvider } from './pages'
 
 ReactDOM.render(
   <React.StrictMode>
-    <Routes />
+    <PlanetProvider>
```
Comment (xXValhallaCoderXx) [inline packages/client/src/index.js:10]: If you wanted to keep the context closer to the app we could have "Page related routes" within `pages` Thats where I would usually put nested routes anyway

I.E
/pages/planet/index.js

```
<PlanetProvider>
<Switch>
    <Route path="/">
         <Planet />
    </Route>
    <Route path="/something/nested">
         <PlanetNested />
    </Route>
</Switch>
</PlanetProvider>
```

File: packages/client/src/pages/planet/planet.contex.js [hunk 2]
```diff
@@ -0,0 +1,108 @@
...
+          planet: null
+        }
+      }
+
+    default:
+      break
+  }
+}
+
+// the reducer context with logging
```
Comment (xXValhallaCoderXx) [inline packages/client/src/pages/planet/planet.contex.js:50]: Im game to use this!
Though would have thought we could have used a library of some sort as it has it all baked in for us,

But either way I dont really care as long as it works :D

- would it be possible we made these into factory functions of some sort? So we dont need to write so much each time we make a new context?

File: packages/client/src/shared/services/http.js [hunk 3]
```diff
@@ -0,0 +1,60 @@
+export async function postRequest (url, payload) {
```
Comment (xXValhallaCoderXx) [inline packages/client/src/shared/services/http.js:1]: Curious, could we not just make 1 function? and just pass in the `method` and payload is optional?
They all seem the same? 

Though im not really bothered - just a thought ^_^

File: packages/client/src/pages/planet/planet.contex.js (hunk 2 above)
Comment (xhad) [reply packages/client/src/pages/planet/planet.contex.js:50]: I would love to have some factory functions l! Lets do it! but I wouod prefer not having any state logic in libraries for security reasons

File: packages/client/src/index.js (hunk 1 above)
Comment (xhad) [reply packages/client/src/index.js:10]: Yes I agree. Lets make this change

File: packages/client/src/shared/services/http.js (hunk 3 above)
Comment (xhad) [reply packages/client/src/shared/services/http.js:1]: Yep I like!

**Pull request merged**